de-lib/src/main/java/org/iplantc/de/client/services/impl/DiskResourceServiceFacadeImpl.java

```diff
@@ -1041,6 +1042,48 @@ public String downloadTemplate(String templateid) {
         return address;
     }
 
+    @Override
+    public void requestHTPathlistFile(HTPathListRequest request, DECallback<File> callback) {
+        StringBuilder address =
+                new StringBuilder(deProperties.getDataMgmtBaseUrl() + "path-list-creator?");
+        address.append("dest=" + URL.encodeQueryString(request.getDest()));
+        if(!Strings.isNullOrEmpty(request.getPattern())) {
+            address.append("&name-pattern=" + URL.encodeQueryString(request.getPattern()));
+        }
+        address.append("&folders-only=" + request.isFoldersOnly());
+        address.append("&recursive=" + true);
+        if (request.getInfoTypes() != null && request.getInfoTypes().size() > 0) {
+            for (String infotype : request.getInfoTypes()) {
+                address.append("&info-type=" + infotype);
+            }
+        }
+        Splittable s = AutoBeanCodex.encode(AutoBeanUtils.getAutoBean(request));
+        Splittable spath = StringQuoter.createSplittable();
+        s.get("paths").assign(spath,"paths");
```

Comment:
Hm... would it make more sense to have them separate then? The `ToolbarViewPresenter` would get the request from the dialog, then call a different method to get the paths.  The presenter would then put the paths in a `HasPaths` object.  Then, the service call would accept the `HtPathListRequest`, `HasPaths`, and the callback.  Then you don't have to do any manipulation here in the service facade.